src/System.Text.Json/src/System/Text/Json/Serialization/JsonNameAttribute.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// Specifies a property name.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+    public sealed class JsonNameAttribute : JsonAttribute
+    {
+        public JsonNameAttribute()
+        {
+        }
+
+        public JsonNameAttribute(string name)
+        {
+            Name = name;
+        }
+
+        public string Name { get; set; }
+    }
```

Comment:
I (oddly) missed that there was an API review. 🤷‍♂️ 

What as the reason why we allow `Name` to be set via ctor and property, and allowed to be null here? This attribute seems to imply that the following are all valid:

```C#
[JsonName]
[JsonName("Foo")]
[JsonName(Name = "Foo")]
```

Reply:
I was going to say something but I assume the effect of placing `[JsonName]` on a property would be to exclude it from being changed by the naming policy.

Reply:
That is what we had in the suggested API.

However I do like the thought of using just `[JsonName]` to indicate use the current name without changing\using other policies. Currently that would be a run-time exception.

Reply:
Changing semantics to null name == use existing name (unless I hear otherwise).

Reply:
I'm a little iffy about this. If you want to use the name of the property verbatim, then you can just write that. Allowing null in through the API and ascribing a new semantic to it ("null means ignore name policy") seems more clever than necessary.

There's an appeal to having the name be required on the attribute because it's bascially impossible to misuse.

Reply:
So the expected semantics:
- Don't expose empty ctor on attribute
- Throw (on first (de)serialization) if null is returned from policy or specified by attribute.